02.calendar/cal.js

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env node
+import minimist from "minimist";
+
+const HEADER_WIDTH = 14;
+
+const run = () => {
+  const today = new Date();
+  const currentYear = today.getFullYear();
+  const currentMonth = today.getMonth() + 1;
+
+  const parseOptions = minimist(process.argv.slice(2), {
+    string: ["-y", "-m"],
+    alias: {
+      y: "year",
+      m: "month",
+    },
+    default: {
+      y: currentYear,
+      m: currentMonth,
+    },
+  });
+
+  const year = parseOptions.year;
+  const month = parseOptions.month;
+
+  return [
+    generateCalendarHeader(year, month),
+    "Su Mo Tu We Th Fr Sa",
+    ...generateMonthDates(year, month),
+  ].join("\n");
+};
+
+const generateCalendarHeader = (year, month) => {
+  const monthName = getFirstDateOfMonth(year, month).toLocaleString("en-US", {
+    month: "short",
+  });
+  return `${monthName} ${year}`.padStart(HEADER_WIDTH, " ");
+};
+
+const getFirstDateOfMonth = (year, month) => {
+  return new Date(year, month - 1, 1);
+};
+
+const generateMonthDates = (year, month) => {
+  const lastDay = getLastDayOfMonth(year, month);
+  const weeks = [];
+  let monthDates = "";
+  monthDates += generateBlankDays(year, month);
+
+  for (let day = 1; day <= lastDay; day++) {
+    monthDates += generateDayFormat(day);
+    if (isSaturday(year, month, day) || day === lastDay) {
+      weeks.push(monthDates);
+      monthDates = "";
+    } else {
+      monthDates += " ";
+    }
+  }
+  return weeks;
+};
+
+const getLastDayOfMonth = (year, month) => {
+  return new Date(year, month, 0).getDate();
+};
+
+const generateBlankDays = (year, month) => {
+  return "   ".repeat(getFirstDateOfMonth(year, month).getDay());
+};
+
+const generateDayFormat = (day) => {
+  return day.toString().padStart(2, " ");
```

Comment:
```suggestion
  return day.toString().padStart(2);
```

デフォルトが空白なので引数を省略してもいいかもしれません

https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Reply:
検収後にもかかわらずレビューありがとうございます😄
リファレンスを確認してみたらその旨の記載がありました📝

- padString
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/String/padStart#padstring(省略可)
> 現在の文字列を延長するための文字列です。padString が targetLength より長い場合は、右からその長さ分が切り捨てられます。 `この引数の既定値は、Unicode の「空白」文字 (U+0020) です`。

引数を省略できるのは知らなかったです😲
（メソッドのリファレンスは使用する度にしっかり読み込みたいと思いました💡）
他にもpadStartを使用している箇所があるので、これらもご提案の内容に修正したいと思います〜